database_wrapper.py

```diff
@@ -599,18 +596,6 @@ def save_audio_sample_meta_data(self, formatted_data: dict) -> bool:
         Returns:
             True if all is good, else False
         """
```

Comment:
About `save_audio_sample_meta_data`: the reason it's separate from `save_entity` is because the keys/values in the data_dictionary don't match the names of the Entity fields. (ex: isWakeWord = "ww" versus metadata.is_wake_word = True) Here are some possible solutions:

1. We leave it as is in this commit - ie: separate `save_entity` and `save_audio_sample_meta_data` functions
    * Pros: Don't have to add extra logic or parameters to `save_entity`
    * Cons What's the point of a generalized function if you have to handle edge cases? 
      * 💩 c 💩 o 💩 d 💩 e 💩 s 💩 m 💩 e 💩 l 💩 l 💩

2. Run a formatting method of some kind on the "raw" data dictionary before it's passed into `save_entity`
    * Pros: `save_entity` is completely generic now
    * Cons: Caller of `save_entity` must know when and how to format the data dictionary before passing it into `save_entity`

ex:
```
# Caller handles when to run format_data
# So the format_data(data_dict) is optional
formatted_data = format_data(data_dict)
save_entity(EntityType, formatted_data, fields=[whatever])
```

3. Have a mapping from Entity types to formatting methods, and run the method (if there is one) on the "raw" data dictionary that's passed into `save_entity`, before we actually populate the Entity.
    * Pros: (IMO) the cleanest solution
    * Cons: ???

ex:
```
def save_entity(EntityType, data_dict, fields=[whatever])
    # This will map Entity Class types to Callables
    FORMAT_BY_ENTITY = {
        AudioSampleMetaData : format_audio_sample_meta_data_dict,
        OtherEntity : format_other_data_dict,
        ...                 : ...
    }

    formatted_dict = FORMAT_BY_ENTITY[EntityType](data_dict)
    # rest of code

# Now, we only need to call this to save an Entity:
save_entity(EntityType, data_dict, fields=[whatever])
# Super generic, super clean, super easy
```

Please let me know what you think is best!

Reply:
One other solution that I forgot to mention: we can pass a formatting method into `save_entity` and run it on the data_dict

Reply:
Ex: 
```
def save_entity(EntityType, data_dict, fields=[whatever], formatter=None):
    formatter(data_dict)
    ....
```